spawn.lua

```diff
@@ -1,66 +1,70 @@
+local Str = {
+	SpawnBeingSend = "You are sent back to %s's spawn",
+	SpawnSentOther = "You sent %s back to %s's spawn",
+	PlayerNotFound = "Player %s not found!",
+	PermissionMissingSpecific = "You need %s permission to do that!",
+	SpawnReached = "Returned to %s's spawn",
+	SpawnSet = "Changed spawn position of %s to [X:%i Y:%i Z:%i]",
+	SpawnSetFailure = "Couldn't change spawn position of %s to [X:%i Y:%i Z:%i]"
+}
+
+local function SendToWorldSpawn( Player, Callback )
+
+	local World = Player:GetWorld()
+	local Chunks = {}
+	table.insert( Chunks, { math.floor(World:GetSpawnX() / 16), math.floor(World:GetSpawnZ() / 16) } )
+	local OnAllChunksAvailable = function()
+		Player:TeleportToCoords(World:GetSpawnX(), World:GetSpawnY(), World:GetSpawnZ())
```

Comment:
Can't use `Player` in the ChunkStay callback, the object could have been deleted in the meantime (player disconnect etc.).

You need to store the player's name or uuid and re-query the object from the world again with `cWorld:DoWithPlayerBy...`